apis/bigqueryconnection/v1alpha1/connection_reference.go

```diff
@@ -120,37 +120,37 @@ func (r *BigQueryConnectionConnectionRef) Parent() (string, error) {
 	return "", fmt.Errorf("BigQueryConnectionConnectionRef not normalized to External form or not created from `New()`")
 }
 
-// NormalizeToExternalForm provision the "External" value.
+// NormalizedExternal provision the "External" value.
 // If the "External" comes from the ConfigConnector object, it has to acquire or reconcile with the GCP resource already.
-func (r *BigQueryConnectionConnectionRef) NormalizeToExternalForm(ctx context.Context, reader client.Reader) error {
+func (r *BigQueryConnectionConnectionRef) NormalizedExternal(ctx context.Context, reader client.Reader, othernamespace string) (string, error) {
```

Comment:
Ah, so I think the reason users get confused is because the `External` field is public, thus we cannot guard the `External` field to be called after the `NormalizedExternal` from the code level.  So if we change the public `External` to private `external`, and only have users call the `NormalizedExternal` function to get the External value (what I'm doing here without changing the External), this confusion can be cleared. Though, I think we don't want to change the `External` type. 

About "whether we are modifying r or not" confusion, users may not care that much when using the function because the when calling a reference, the External is only used in the single `_ToProto` method (where they should use `NormalizedExternal`).